src/main.rs

```diff
@@ -138,8 +138,12 @@ fn inner_main() -> Result<ExitCode> {
     }
 
     // Initialize the cache.
+    let cache_dir = match &pyproject_strategy {
+        PyprojectDiscovery::Fixed(settings) => &settings.cache_dir,
+        PyprojectDiscovery::Hierarchical(settings) => &settings.cache_dir,
+    };
     let mut cache_enabled: bool = !cli.no_cache;
-    if cache_enabled && cache::init().is_err() {
+    if cache_enabled && cache::init(cache_dir).is_err() {
```

Comment:
I think we may need to call `cache::init` for every `Settings` object now, since different files could end up using different cache directories.

Reply:
Not sure how to go about this. Was thinking at first to do that in `commands::run` before running `lint_path`. But not only would that then happen per file, it can also happen at the same time due to `par_iter`.

My only other idea is to initialize the cache at the beginning for all resolved settings, keeping track of that and checking that per file.

Reply:
Not sure how to go about this. Was thinking at first to do that in commands::run before running lint_path. But not only would that then happen per file, it can also happen at the same time due to par_iter.

My only other idea is to initialize the cache at the beginning for all resolved settings, keeping track of that and checking that per file.

Reply:
I think the latter option probably makes sense, to initialize for all resolved settings once we’ve gathered them up.

Reply:
Conceptually it’d be nice to just do it in the cache setter (like, right before we write), but maybe that’s hard to get right in a multi-threaded context?